pennylane/beta/tapes/qnode.py

```diff
@@ -97,6 +98,10 @@ class QNode:
             * ``"finite-diff"``: Uses numerical finite-differences for all quantum operation
               arguments.
 
+        caching (int): number of device executions to store in a cache to speed up subsequent
+            executions. Caching does not take place by default. In caching mode, the quantum circuit
+            being executed must have a constant structure and only its parameters can be varied.

```

Comment:
In addition (not sure if it's implied by 'only its parameters can be varied', but it wasn't clear to me): there can't be any classical processing within the QNode.

E.g., `qml.RX(2*x, wires=0)` is not allowed.

This is a side effect of the tape being a part of the users quantum function --- we can't extract the tape independently of the classical processing :(